### PR TITLE
Networking should be used to hold network related pkgs

### PR DESCRIPTION
**Pull request by resouer**
ref #15634 

cc @lavalamp

### CODE REVIEW

**Comment by resouer** [comment]
@lavalamp Rebased and updated :)

**Comment by lavalamp** [comment]
LGTM, thanks.

File: pkg/util/networking/http.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package util
+package networking
```
Comment (thockin) [inline pkg/util/networking/http.go]: It's a minor point but `util/net` would be shorter :)

File: pkg/util/networking/http.go [hunk 2]
```diff
@@ -102,3 +102,12 @@ func TLSClientConfig(transport http.RoundTripper) (*tls.Config, error) {
 		return nil, fmt.Errorf("unknown transport type: %v", transport)
 	}
 }
+
+func GetClient(req *http.Request) string {
```
Comment (thockin) [inline pkg/util/networking/http.go]: GetHTTPClient?

File: pkg/util/networking/interface.go [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+	if finalIP == nil {
+		return nil, fmt.Errorf("Unable to select an IP.")
+	}
+	return nil, nil
+}
+
+// If bind-address is usable, return it directly
+// If bind-address is not usable (unset, 0.0.0.0, or loopback), we will use the host's default
+// interface.
+func ValidPublicAddrForMaster(bindAddress net.IP) (net.IP, error) {
```
Comment (thockin) [inline pkg/util/networking/interface.go]: ChooseBindAddress() ?  This is on the hairy edge of useful.

**Comment by thockin** [comment]
Thanks!  Just small points from me.

**Comment by resouer** [comment]
@k8s-bot e2e test this

File: pkg/util/networking/http.go (hunk 1 above)
Comment (resouer) [inline pkg/util/networking/http.go]: Em, I hope to keep this name because  this pkg offen appears with `net`.

Comment (resouer) [inline pkg/util/networking/http.go]: Aha, seems not too much conflicts, I wrote a shell to help me refactor this, gonna update soon :)

**Comment by resouer** [comment]
@lavalamp @thockin Done! Request for green labels :)

File: cmd/integration/integration.go [hunk 4]
```diff
@@ -53,6 +53,7 @@ import (
 	"k8s.io/kubernetes/pkg/labels"
 	"k8s.io/kubernetes/pkg/master"
 	"k8s.io/kubernetes/pkg/util"
+	networking "k8s.io/kubernetes/pkg/util/net"
```
Comment (thockin) [inline cmd/integration/integration.go]: convention would call this utilnet

**Comment by thockin** [comment]
just a nit on the import aliases and LGTM.

Thanks for this!

**Comment by resouer** [comment]
@thockin Updated :)

**Comment by resouer** [comment]
@thockin Fixed rebase, please add label

**Comment by resouer** [comment]
@k8s-bot unit test this please

**Comment by resouer** [comment]
@k8s-bot unit test this.  This is wired, am I running into a flaky integration test?

**Comment by alex-mohr** [comment]
@k8s-bot unit test this

**Pull request merged**